test/dandelion_voting_disputable.js

```diff
@@ -0,0 +1,373 @@
+      assert.isFalse(closed, 'action is not closed')
+    })
+
+    it('canChallenge returns true', async () => {
+      assert.isTrue(await voting.canChallenge(voteId))
+    })
+
+    it('canClose returns false', async () => {
+      assert.isFalse(await voting.canClose(voteId))
+    })
+  })
+
+  describe('execute', () => {
+    beforeEach(async () => await createVote({ voter: voter51, cast: true, execute: true }))
+
+    it('changes the disputable state to closed', async () => {
+      const { actionId: voteActionId, pausedAtBlock, pauseDurationBlocks, status } = await voting.getDisputableInfo(voteId)
+      assertBn(status, VOTE_STATUS.CLOSED, 'vote status does not match')
+
+      assertBn(voteActionId, actionId, 'action ID does not match')
+      assertBn(pausedAtBlock, 0, 'paused at does not match')
+      assertBn(pauseDurationBlocks, 0, 'pause duration does not match')
+    })
+
+    it('closes the action on the agreement and executed the vote', async () => {
+      assertBn(await executionTarget.counter(), 1, 'vote was not executed')
+
+      const { disputable, disputableActionId, collateralRequirementId, context, closed, submitter } = await agreement.getAction(actionId)
+
+      assert.isTrue(closed, 'action is not closed')
+      assertBn(disputableActionId, voteId, 'disputable ID does not match')
+      assert.equal(disputable, voting.address, 'disputable address does not match')
+      assertBn(collateralRequirementId, 1, 'collateral ID does not match')
+      assert.equal(toAscii(context), 'metadata', 'context does not match')
+      assert.equal(submitter, voter51, 'action submitter does not match')
+    })
+
+    it('canChallenge returns false', async () => {
+      assert.isFalse(await voting.canChallenge(voteId))
+    })
+
+    it('canClose returns true', async () => {
+      assert.isTrue(await voting.canClose(voteId))
+    })
+  })
+
+  describe('challenge', () => {
+    let challengeBlockNumber
+
+    beforeEach(async () => {
+      await createVote({ voter: voter51, cast: false })
+      await agreement.challenge({ actionId })
+      challengeBlockNumber = await voting.getBlockNumberPublic()
+    })
+
+    it('pauses the vote', async () => {
+      const { actionId: voteActionId, pausedAtBlock, pauseDurationBlocks, status } = await voting.getDisputableInfo(voteId)
+      assertBn(status, VOTE_STATUS.PAUSED, 'vote status does not match')
+
+      assertBn(voteActionId, actionId, 'action ID does not match')
+      assertBn(pausedAtBlock, challengeBlockNumber, 'paused at does not match')
+      assertBn(pauseDurationBlocks, 0, 'pause duration does not match')
+    })
+
+    it('does not allow a voter to vote', async () => {
+      assert.isFalse(await voting.canVote(voteId, voter49), 'voter can vote')
+
+      await assertRevert(voting.vote(voteId, false, { from: voter49 }), 'DANDELION_VOTING_CANNOT_VOTE')
+    })
+
+    it('does not allow to execute the vote', async () => {
```

Comment:
We could also add a test for when the vote has reached required support and quorum and the execution time has passed

Reply:
Oops, I made a bug, good one. Could have executed challenged votes if not for this.